core/new-gui/src/app/workspace/component/navigation/navigation.component.ts

```diff
@@ -295,8 +309,8 @@ export class NavigationComponent implements OnInit {
   /**
    * Handler for changing workflow name input box, updates the cachedWorkflow and persist to database.
    */
-  onWorkflowNameChange() {
-    this.currentWorkflowName = this.currentWorkflowName.trim();
+  onWorkflowNameChange(event: InputEvent) {
+    this.currentWorkflowName = (<HTMLElement>event.target).innerText.trim();
```

Comment:
why do we need to use native HTML elements to do this?

Reply:
See the two-way data binding comment in the corresponding html element

Reply:
I think this is a bug with Angular:
two-way binding to content just doesn't work with the contenteditable="true" attribute. Editing content doesn't trigger the event listener to update this.currentWorkflowName. Manually triggering the event listener still works though.

To fix this, I'd need to use an <input> tag. However <input> tags don't expand when you type into them, and I can't replicate that effect without some fancy JS